### PR TITLE
Update ReorderableListView API docs

### LINKED ISSUE #69482: Adopt a Widget - ReorderableListView
The [ReorderableListView](https://master-api.flutter.dev/flutter/material/ReorderableListView-class.html) widget needs:

- A code / DartPad sample

For details about how to participate, see [flutter.dev/adoptawidget](https://flutter.dev/adoptawidget)

### PR DESCRIPTION
**Pull request by sanjoli63**
AdoptAWidget :  ReorderableLIstView

This pull request is:

- [x] A code sample
- [ ] More references
- [ ] More explanation

closes #69482

### CODE REVIEW

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 1]
```diff
@@ -66,14 +103,14 @@ class ReorderableListView extends StatefulWidget {
...
     this.reverse = false,
-  }) : assert(scrollDirection != null),
-       assert(onReorder != null),
-       assert(children != null),
-       assert(
-         children.every((Widget w) => w.key != null),
-         'All children of this widget must have a key.',
-       ),
-       super(key: key);
+  })  : assert(scrollDirection != null),
```
Comment (Abhishek01039) [inline packages/flutter/lib/src/material/reorderable_list.dart:106]: @sanjoli63 Please discard these changes and save a file without formatting.
Thanks.

**Review by Abhishek01039 — CHANGES_REQUESTED** [review]

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 2]
```diff
@@ -54,6 +54,45 @@ typedef ReorderCallback = void Function(int oldIndex, int newIndex);
...
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=3fB1mxOsqJE}
+///
+/// This sample shows by dragging the user can reorder the items of the list.
+/// list of Numbers will be used to display items in listview.
+/// [generate]it is constructor which  is used to form a list of values.
+/// [onReorder] This parameter is compulsory. It will be called when list child
+/// is moved to a new position.
+///
+/// {@tool dartpad --template=stateless_widget_scaffold}
```
Comment (Amitpatil215) [inline packages/flutter/lib/src/material/reorderable_list.dart:64]:
```suggestion
/// {@tool dartpad --template=stateful_widget_scaffold}
```

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 3]
```diff
@@ -54,6 +54,45 @@ typedef ReorderCallback = void Function(int oldIndex, int newIndex);
 /// All [children] must have a key.
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=3fB1mxOsqJE}
+///
+/// This sample shows by dragging the user can reorder the items of the list.
+/// list of Numbers will be used to display items in listview.
+/// [generate]it is constructor which  is used to form a list of values.
```
Comment (PixelToast) [inline packages/flutter/lib/src/material/reorderable_list.dart:60]: The `[generate]` here is causing an unresolved doc reference error

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 4]
```diff
@@ -54,6 +54,45 @@ typedef ReorderCallback = void Function(int oldIndex, int newIndex);
...
+/// {@tool dartpad --template=stateful_widget_scaffold}
+///
+/// ```dart
+/// List<String> _list = List.generate(5, (i) => "${i}");
+///
+/// Widget build(BuildContext context){
+///   return ReorderableListView(
+///     padding : const EdgeInsets.symmetric(horizontal:40),
+///     children: _list.map((String string) => ListTile(
+///       key: Key(_list[_list.indexOf(string)]),
```
Comment (johnpryan) [inline packages/flutter/lib/src/material/reorderable_list.dart:73]: It's probably better to base the key off of `_list.indexOf(string)` instead of the value at that location. This section could also be simplified a little by using a *collection for*:

```dart
      children: [
        for (var i = 0; i < _list.length; i++)
          ListTile(key: Key('$i'), title: Text(_list[i])),
      ],
```

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 5]
```diff
@@ -54,6 +54,45 @@ typedef ReorderCallback = void Function(int oldIndex, int newIndex);
...
+///         }
+///         _list[newIndex]= old;
+///       }
+///       else{
+///         for(int i = oldIndex; i< newIndex-1;i++){
+///           _list[i]= _list[i+1];
+///         }
+///         _list[newIndex -1] = old;
+///       }
+///       setState((){});
```
Comment (johnpryan) [inline packages/flutter/lib/src/material/reorderable_list.dart:89]: I would use the same technique described in [ReorderCallback's API docs](https://api.flutter.dev/flutter/material/ReorderCallback.html):

```dart
      onReorder: (oldIndex, newIndex) {
        setState(() {
          if (oldIndex < newIndex) {
            // removing the item at oldIndex will shorten the list by 1.
            newIndex -= 1;
          }

          final element = _list.removeAt(oldIndex);
          _list.insert(newIndex, element);
        });
      },
```

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 6]
```diff
@@ -54,6 +54,45 @@ typedef ReorderCallback = void Function(int oldIndex, int newIndex);
 /// All [children] must have a key.
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=3fB1mxOsqJE}
+///
+/// This sample shows by dragging the user can reorder the items of the list.
+/// list of Numbers will be used to display items in listview.
```
Comment (johnpryan) [inline packages/flutter/lib/src/material/reorderable_list.dart:59]: "in listview" -> "in the list"

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 7]
```diff
@@ -54,6 +54,45 @@ typedef ReorderCallback = void Function(int oldIndex, int newIndex);
...
+/// [List.generate]it is constructor which  is used to form a list of values.
+/// [onReorder] This parameter is compulsory. It will be called when list child
```
Comment (johnpryan) [inline packages/flutter/lib/src/material/reorderable_list.dart:61]: "[onReorder] This parameter is compulsory. It will be called when list child is moved to a new position" -> "The [onReorder] parameter is required and will be called when a child widget is dragged to a new position."

**Review by johnpryan — CHANGES_REQUESTED** [review]

**Comment by sanjoli63** [comment]
@googlebot I fixed it

**Comment by sanjoli63** [comment]
@googlebot I fixed it.

**Comment by johnpryan** [comment]
@sanjoli63 It looks like you need to merge with the latest `master` branch

**Comment by johnpryan** [comment]
I would reword this paragraph:

```
/// This sample shows by dragging the user can reorder the items of the list.
/// list of Numbers will be used to display items in the list.
/// [List.generate]it is constructor which  is used to form a list of values.
/// The [onReorder] parameter is required and will be called when a child
/// widget is dragged to a new position.
```

to

```
/// This sample shows that by dragging the user can reorder the items of the list.
/// The [onReorder] parameter is required and will be called when a child
/// widget is dragged to a new position.
```

**Comment by johnpryan** [comment]
I would also consider adding a comma after the `title` parameter in `ListTile` and reformatting:

```
  Widget build(BuildContext context) {
    return ReorderableListView(
      padding: const EdgeInsets.symmetric(horizontal: 40),
      children: [
        for (var i = 0; i < _list.length; i++)
          ListTile(
            key: Key('$i'),
            title: Text(_list[i]),
          ),
      ],
      onReorder: (oldIndex, newIndex) {
        setState(() {
          if (oldIndex < newIndex) {
            newIndex -= 1;
          }
          final element = _list.removeAt(oldIndex);
          _list.insert(newIndex, element);
        });
      },
    );
  }
```

**Comment by johnpryan** [comment]
@Amitpatil215, could you sign the CLA?

**Comment by Amitpatil215** [comment]
> @Amitpatil215, could you sign the CLA?

Yes, Signed earlier
<details><summary>Refer Snapshot</summary>
<p>

Screenshot (62)

</p>
</details>

**Comment by johnpryan** [comment]
@Amitpatil215 thanks, you're right, I can confirm that you signed it.

**Comment by johnpryan** [comment]
For CLA bot: this was an issue with verifying a suggestion from 54329870+Amitpatil215@users.noreply.github.com, but the author has signed.

**Review by johnpryan — APPROVED** [review]

**Comment by johnpryan** [comment]
@sanjoli63 Thanks for participating in AdoptAWidget. We would like to send you a special thanks. If you are interested, please send an email to nyener at google dot com and provide a link to this issue.
